fe/fe-core/src/main/java/com/starrocks/catalog/SchemaInfo.java

```diff
@@ -0,0 +1,270 @@
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public short getShortKeyColumnCount() {
+        return shortKeyColumnCount;
+    }
+
+    public KeysType getKeysType() {
+        return keysType;
+    }
+
+    public TStorageType getStorageType() {
+        return storageType;
+    }
+
+    public int getVersion() {
+        return version;
+    }
+
+    public List<Column> getColumns() {
+        return columns;
+    }
+
+    public List<Integer> getSortKeyIndexes() {
+        return sortKeyIndexes;
+    }
+
+    public List<Integer> getSortKeyUniqueIds() {
+        return sortKeyUniqueIds;
+    }
+
+    public List<Index> getIndexes() {
+        return indexes;
+    }
+
+    public Set<String> getBloomFilterColumnNames() {
+        return bloomFilterColumnNames;
+    }
+
+    public double getBloomFilterFpp() {
+        return bloomFilterFpp;
+    }
+
+    public TTabletSchema toTabletSchema() {
+        TTabletSchema tSchema = new TTabletSchema();
+        tSchema.setShort_key_column_count(shortKeyColumnCount);
+        tSchema.setKeys_type(keysType.toThrift());
+        tSchema.setStorage_type(storageType);
+        tSchema.setId(id);
+        tSchema.setSchema_version(version);
+        tSchema.setSchema_hash(schemaHash);
+        tSchema.setIs_in_memory(false/*unused now*/);
+
+        List<TColumn> tColumns = new ArrayList<TColumn>();
+        for (Column column : columns) {
+            TColumn tColumn = column.toThrift();
+            // is bloom filter column
+            if (bloomFilterColumnNames != null && bloomFilterColumnNames.contains(column.getName())) {
+                tColumn.setIs_bloom_filter_column(true);
+            }
+            // when doing schema change, some modified column has a prefix in name.
+            // this prefix is only used in FE, not visible to BE, so we should remove this prefix.
+            if (column.getName().startsWith(SchemaChangeHandler.SHADOW_NAME_PRFIX)) {
+                tColumn.setColumn_name(column.getName().substring(SchemaChangeHandler.SHADOW_NAME_PRFIX.length()));
+            }
+            if (column.getName().startsWith(SchemaChangeHandler.SHADOW_NAME_PRFIX_V1)) {
+                tColumn.setColumn_name(column.getName().substring(SchemaChangeHandler.SHADOW_NAME_PRFIX_V1.length()));
+            }
+            tColumns.add(tColumn);
+        }
+        tSchema.setColumns(tColumns);
+        tSchema.setSort_key_idxes(sortKeyIndexes);
+        tSchema.setSort_key_unique_ids(sortKeyUniqueIds);
+
+        if (CollectionUtils.isNotEmpty(indexes)) {
+            List<TOlapTableIndex> tIndexes = new ArrayList<>();
+            for (Index index : indexes) {
+                tIndexes.add(index.toThrift());
+            }
+            tSchema.setIndexes(tIndexes);
+        }
+
+        if (bloomFilterColumnNames != null) {
+            tSchema.setBloom_filter_fpp(bloomFilterFpp);
+        }
+        return tSchema;
+    }
+
+    public static Builder builder() {
```

Comment:
can name to `newBuilder()` referring to protobuf Builder, `builder` interface is a little close to `build()`.

Reply:
Updated